### DISCUSSION TITLE
Data issue on Vertix 2 - significantly impacting alphas

### DISCUSSION THREAD

**Discussion by Logiqx** [djong1/gp3s#173] [GPS Technology (devices and formats)]
I've created this discussion to highlight an issue that may affect multiple people / sessions.

A friend of mine (Will Trossell) posted a session ([403861](https://www.gps-speedsurfing.com/default.aspx?mnu=user&val=403861&uid=2911&spotid=598)) with an exceptionally good alpha which was queried by friends as it is a new spot record; 27.75 knots.

Will sent me the FIT file and long story cut short, the best alpha is probably around 18 knots and not >27 knots. This has occurred because the positional data is lagging the speed data by about 55 seconds-ish.

This can be seen in GPSResults; doppler-speed (red) vs position-derived speed (green).

![image](https://user-images.githubusercontent.com/17449078/231568348-8fb49d2a-af53-4fe2-9a00-0c3b84fa2cc4.png)

Algorithms that calculate alpha results typically use positional data to determine the start / end points and the speed data (typically Doppler) to calculate the result. Since they are 55 seconds apart the alpha results are based on speeds 55 seconds before the actual alpha!

I haven't seen this in COROS data before but it's clearly something that needs further investigation.

I will ask Will for some more tracks and see if he knows what firmware was being used at the time.

**Comment by Logiqx** [comment]
Firmware was 2.86.0 - 10 Mar 2023

A session from the previous day used the same firmware but did not have this issue.

I don't know if it is a one-off or potentially affects other COROS users but the cause is likely to be a COROS bug (imho).

**Comment by Logiqx** [comment]
Further information from Will. The issue observed above was on his current VERTIX 2 which was his second VERTIX 2.

The issue also happened with his first VERTIX 2 (replaced under warranty, dodgy button), during a session on 8 Aug 2022. The older VERTIX 2 was also using an earlier firmware.

The session from Aug 2022 consisted of 3 parts. Part 1 was fine but parts 2 and 3 had the shifting issue.

Parts 1 and 2:

![image](https://user-images.githubusercontent.com/17449078/232054042-3683bb6d-19de-4b0f-998b-c8e1d5a68f77.png)

Parts 2 and 3:

![image](https://user-images.githubusercontent.com/17449078/232054146-fb1b39a6-f551-45a2-a804-69c1cb3fff0a.png)

So, this confirms an issue that affects multiple watches (both VERTIX 2, but other models may also be affected) and multiple firmwares.

My suspicion is that it either occurs when the watch syncs with the phone app, or subsequent processing on the COROS servers.

This will need raising with COROS.

**Comment by Logiqx** [comment]
A few sessions affected.

- https://www.gps-speedsurfing.com/default.aspx?mnu=user&val=405404&uid=2911 (definite issue, not APEX 2 Pro as shown)
- https://www.gps-speedsurfing.com/default.aspx?mnu=user&val=391113&uid=2911 (definite, VERTIX 2)
- https://www.gps-speedsurfing.com/default.aspx?mnu=user&val=391501&uid=15697 (maybe)

**Reply by jhdebruin** [reply]
First two have been adjusted, the third has been flagged now. For future cases please also use the FLAG option, so we can keep the standard procedure for this.  @djong1 can this be caused by splitting the sessions caused by longer time-outs ?

**Reply by Logiqx** [reply]
@jhdebruin - It's not something caused by GP3S. It's definitely a data issue in the COROS files.

**Reply by jhdebruin** [reply]
As discussed , we received the .FIT file from the specific rider and its confirmed that also this session is affected. Further action will be taken towards Coros

**Comment by Logiqx** [comment]
I have just realised that I never added this item to the COROS issue list.

Now added as issue 25 in my [gp3s-coros](https://github.com/Logiqx/gp3s-coros/issues/25) repo. It links back to this discussion for full details.
